eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/serialization/core/Parameter.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2017 SchedJoules
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.schedjoules.eventdiscovery.framework.serialization.core;
+
+/**
+ * @author Gabor Keszthelyi
+ */
+public interface Parameter<T>
+{
+
+    void into(T container);
```

Comment:
I really like your idea to not mutate the container but return a modified copy! That was a basic problem here..
And I see there are copy constructors for both `Intent` and `Bundle`.

I am not sure though how well the above, combined `Parametrized<T>` would work..  
Wouldn't it work well only if we implement it once? For example, please check current `BundleParameter` and `ParcelableBundleParameter`. The 2nd one has input params `Key` and `Parcelable`, not `Key` and `Box`.
So I am wondering if it's better to keep them separate, the common implementations for the actual parameter entry and the builder.
So what do you think about this?:
```
interface Param<C>
{
    C updated(C originalCarrier);
}

interface FluentParam<C> extends Param<C>
{
    <T> FluentParam withParam(Key<T> key, Box<T> box);
}
```
Names can change of course. Maybe no need for the general `Fluent*`, we can have specific for `Intent`s with `withExtra()` and for `Bundles` with `withArg()`.

Usage could be this:
```
Bundle bundle = new BundleBuilder()
    .withParam(key1, box1)
    .withParam(key2, box2)
    .updated(new Bundle());

or for single param:
Bundle bundle = new BundleParam(key, box).updated(new Bundle());
Bundle bundle = new ParcelableBundleParam(key, parc).updated(new Bundle());
```